ec2/ec2test/server.go

```diff
@@ -1955,13 +2037,26 @@ func (srv *Server) describeIFaces(w http.ResponseWriter, req *http.Request, reqI
 	resp.XMLName = xml.Name{defaultXMLName, "DescribeNetworkInterfacesResponse"}
 	resp.RequestId = reqId
 	for _, i := range srv.ifaces {
-		ok, err := f.ok(i)
-		_, known := idMap[i.Id]
-		if ok && (len(idMap) == 0 || known) {
+		// Without neither filers nor ids - return everything.
+		if len(f) == 0 && len(idMap) == 0 {
 			resp.Interfaces = append(resp.Interfaces, i.NetworkInterface)
-		} else if err != nil {
+			continue
+		}
+		filterMatch, err := f.ok(i)
+		if err != nil {
 			fatalf(400, "InvalidParameterValue", "describe ifaces: %v", err)
 		}
+		_, requestedByID := idMap[i.Id]
+		if len(idMap) == 0 {
+			// since filter.ok() returns true when none matches, we
```

Comment:
Looking at it this morning it *does* look confusing :)
I've simplified the code to a single conditional: `if filterMatch && (len(idMap) == 0 || idMap[i.Id] { ...`, which works for all cases. Thanks!